logstash-plugins/logstash-input-azure_event_hubs
# Update Azure event Hub client library to version 3.3.0

## Release notes
Updated Azure Event Hub client library to version 3.3.0

## What does this PR do?
Updates azure-eventhubs to 3.3.0 and all its dependencies. 
- adapt the instantiation of EventProcessorHost using the EventProcessorHostBuilder instead of the now private constructor
- updates the mocking in unit tests to respect the switch from constructor to builder.

## Why is it important/What is the impact to the user?

Permit to use a newer library.

## Checklist


## Author's Checklist


- [x] run against a real Az EH instance and check it can pull data out of the queue.

## How to test this PR locally

- create an Event Hub in Azure portal (https://https://portal.azure.com/)
- in Logstash install a gem generated by this plugin or update the `Gemfile`with:
```
gem "logstash-input-azure_event_hubs", :path => "/path/to/logstash-input-azure_event_hubs"
```
and update with `bin/logstash-plugin install --no-verify`
- run Logstash with a pipeline like:
```
input {
  azure_event_hubs {
    connections => ["Endpoint=sb://<name of the Event Hub>.servicebus.windows.net/;SharedAccessKeyName=test-log;SharedAccessKey=<s3cr3t_1>;EntityPath=test"]
    event_hub_connections => ["Endpoint=sb://<name of the Event Hub>.servicebus.windows.net/;SharedAccessKeyName=eh_policy;SharedAccessKey=<s3cr3t_2>;EntityPath=logstash-event-hub"]
    decorate_events => true
    storage_container => "logstash"
    storage_connection => "DefaultEndpointsProtocol=https;AccountName=<storageaccount>;AccountKey=<s3cr3t_3>;EndpointSuffix=core.windows.net"

  }
}

output {
  stdout{
    codec => rubydebug
  }
}
```
- from Azure EbentHub UI send a message and verify it's processed by Logstash without any error like class or method not found.

## Related issues


-
